### ISSUE TITLE
Doubt regarding aria-labelledby

### ISSUE THREAD

**Issue by ThomasR128** [daisy/kb#49]
In both [example 2](http://kb.daisy.org/publishing/docs/html/headings.html#ex-02) of the Headings KB article and [example 3](http://kb.daisy.org/publishing/docs/html/sections.html#ex-03) of the Sections KB article the `aria-labelledby` references the ID of a `header` element … which doesn’t have any content. Its children of course do.

My understanding was that `aria-labelledby` must always reference an element that itself has content, since its content will be used as the label of the region. Am I mistaken?

Can we rely on AT to parse the header’s children to find one or more elements that do have content, and in these two examples, possibly even combine the `h1` and the `p` into a multi-part label?

**Comment by mattgarrish** [comment]
> My understanding was that `aria-labelledby` must always reference an element that has content

The header element contains both the h1 and the subtitle so there is content there. The [accessible name calculation algorithm](https://www.w3.org/TR/accname-1.1/#mapping_additional_nd_te) defines the rules for parsing and combining element text.

There is a problem with that example, but not how you describe. The name computation excludes the subtitle unless the role is removed. That looks like an implementation bug, as if I switch it to a `header` role, for example, it picks it up again.

For confirmation, here's a screen grab of the computed properties from the chromium accessibility tab with the subtitle role:

<img width="233" alt="name" src="https://user-images.githubusercontent.com/1565164/152229004-f4f07a74-3f4e-4249-8dd5-77bdf9d70c5c.PNG">

And here it is without the subtitle role/with a heading role:

<img width="346" alt="longername" src="https://user-images.githubusercontent.com/1565164/152232594-72c3ac65-2695-47c3-9f5e-7e49f6d61d75.PNG">

I get the same behaviour in NVDA's landmarks list.

**Comment by ThomasR128** [comment]
Matt, 

As always, thanks for your quick reply. I found that Safari on a Mac has the same capability, so in the future I’ll check this kind of issues for myself and will hopefully bother you less 😉. No Windows means no NVDA out-of-the-box, however. Will have to look into a VM.

> The name computation excludes the subtitle unless the role is removed.

Safari gets it right with and without the subtitle role (never mind the incorrect lang tag) – see ‘Etiqueta’:

![Captura de pantalla 2022-02-02 a las 22 00 52](https://user-images.githubusercontent.com/64253366/152237522-946e21f6-970b-449f-93e4-a076711e9170.png)

![Captura de pantalla 2022-02-02 a las 22 01 39](https://user-images.githubusercontent.com/64253366/152237549-9520fa08-696f-48c1-bb61-e90e40992814.png)

If I move the ID to the heading tag, only the `h1` is picked up, no matter if the `p` has a subtitle role or not – as expected. Of course, assigning another ID to the `p` and referencing both results in a correct multi-part label .

![Captura de pantalla 2022-02-02 a las 22 43 56](https://user-images.githubusercontent.com/64253366/152242525-267385f4-9b27-428f-b5cc-0703819f4326.png)

Since I can‘t see your source code I’m not entirely sure if this is what you meant by ‘switch it to a `header` role’ and ‘with a heading role’.

**Comment by mattgarrish** [comment]
> Safari gets it right with and without the subtitle role (never mind the incorrect lang tag)

Ya, that's extra confirmation it's an implementation bug.

> If I move the ID to the heading tag, only the `h1` is picked up, no matter if the `p` has a subtitle role or not – as expected.

Right, because then the section just gets labeled by its heading.

> Since I can‘t see your source code I’m not entirely sure if this is what you meant by ‘switch it to a `header` role’ and ‘with a heading role’.

I was just testing to see if it was an issue caused by the role, so instead of `<p role="doc-subtitle">` I changed the tag to `<p role="heading">` so if it would make things work properly, since both roles inherit from the `sectionhead` abstract class. (I made a typo when I wrote the role name, as it should have been `heading` not `header`.)

**Comment by ThomasR128** [comment]
Understood.
